Contrast bar feature and trimming plotter.py

Jacobluke- opened the pull request.

JAnns98 reviewed (approved):
Significant changes, including trimming the plotter function and creating smaller functions in plot_tools and misc_tools to restructure the code and improve readability and future changes. Some additional edits are needed to add doc strings and comments to these added features in a subsequent patch. 

Currently, the legend for swarmplot with color_col appears to be plotting the legend dots twice in the test images. I am unable to recreate this in my testing notebook, so it seems to be specific to test images - but we should keep this in mind in case it comes up.

Jacobluke- commented:
The latest commit also optimized the add_counts_to_ticks() function by

1. Moved lookup_value function definition outside the loop for better readability.
2. Simplified the lookup_value function by removing redundant exception handling and using a more concise structure.
3. Used str.split('\n')[-1] instead of rfind for better readability and potentially slightly better performance.
4. Removed the unnecessary ticks_loc variable and moved the set_major_locator call to the end of the function.
5. Used plot_kwargs.get("fontsize_rawxlabel") instead of checking if it's not None, which simplifies the code.
